easybuild/easyblocks/f/ferret.py

```diff
@@ -48,80 +50,148 @@ def configure_step(self):
+            (r"^INSTALL_FER_DIR =.*", "INSTALL_FER_DIR = %s" % self.installdir),
+        ]
 
-            line = re.sub(r"^BUILDTYPE\s*=.*", "BUILDTYPE = %s" % buildtype, line)
-            line = re.sub(r"^INSTALL_FER_DIR =.*", "INSTALL_FER_DIR = %s" % self.installdir, line)
+        for name in deps:
+            regex_subs += [(r"^(%s.*DIR\s*)=.*" % name.upper(), r"\1 = %s" % get_software_root(name))]
 
-            for name in deps:
-                line = re.sub(r"^(%s.*DIR\s*)=.*" % name.upper(), r"\1 = %s" % get_software_root(name), line)
+        if LooseVersion(self.version) >= LooseVersion("7.3"):
+            regex_subs += [
+                (r"^DIR_PREFIX =.*", "DIR_PREFIX = %s" % self.cfg['start_dir']),
+                (r"^FER_LOCAL_EXTFCNS = $(FER_DIR)", "FER_LOCAL_EXTFCNS = $(INSTALL_FER_DIR)/libs"),
+            ]
 
-            sys.stdout.write(line)
+        for fn in fns:
+            apply_regex_substitutions(fn, regex_subs)
 
         comp_vars = {
-            'CC':'CC',
-            'CFLAGS':'CFLAGS',
-            'CPPFLAGS':'CPPFLAGS',
-            'FC':'F77',
+            'CC': 'CC',
+            'CFLAGS': 'CFLAGS',
+            'CPPFLAGS': 'CPPFLAGS',
+            'FC': 'F77',
+        }
+
+        gfort2ifort = {
+            '-fno-second-underscore': ' ',
+            '-fno-backslash': ' ',
+            '-fdollar-ok': ' ',
+            '-ffast-math': ' ',
+            '-ffixed-line-length-132': '-132',
+            '-fno-automatic': ' ',
+            '-ffpe-trap=overflow': ' ',
+            '-fimplicit-none': '-implicitnone',
+            '-fdefault-real-8': '-r8',
+            '-fdefault-double-8': ' ',
+            '-Wl,-Bstatic -lgfortran -Wl,-Bdynamic': ' ',
+            '-v --verbose -m64': ' ',
+            '-export-dynamic': ' ',
+            '-DG77_SIGNAL': ' ',
         }
 
         fn = 'xgks/CUSTOMIZE.%s' % buildtype
 
-        for line in fileinput.input(fn, inplace=1, backup='.orig'):
+        regex_subs = [(r"^(FFLAGS\s*=').*-m64 (.*)", r"\1%s \2" % os.getenv('FFLAGS'))]
+        for x, y in comp_vars.items():
+            regex_subs += [(r"^(%s\s*)=.*" % x, r"\1='%s'" % os.getenv(y))]
 
-            for x,y in comp_vars.items():
-                line = re.sub(r"^(%s\s*)=.*" % x, r"\1=%s" % os.getenv(y), line)
+        if LooseVersion(self.version) >= LooseVersion("7.3"):
+            regex_subs += [(r"^(LD_X11\s*)=.*", r"\1='-L%s/lib -lX11'" % get_software_root('X11'))]
+        else:
+            regex_subs += [(r"^(LD_X11\s*)=.*", r"\1='-L/usr/lib64/X11 -lX11'")]
 
-            line = re.sub(r"^(FFLAGS\s*=').*-m64 (.*)", r"\1%s \2" % os.getenv('FFLAGS'), line)
-            line = re.sub(r"^(LD_X11\s*)=.*", r"\1='-L/usr/lib64/X11 -lX11'", line)
+        if LooseVersion(self.version) >= LooseVersion("7.3"):
+            if self.toolchain.comp_family() == toolchain.INTELCOMP:
+                for x, y in gfort2ifort.items():
+                    regex_subs += [(r"%s" % x, r"%s" % y)]
```

Comment:
This can be simplified to:

```python
if LooseVersion(self.version) >= LooseVersion("7.3") and self.toolchain.comp_family() == toolchain.INTELCOMP:
    regex_subs.extend(sorted(gfort2ifort.items()))
```

the `sorted` is only needed to ensure consistent behaviour